<h1>Исследование надёжности заёмщиков<span class="tocSkip"></span></h1>
<ul class="toc-item"><li><span><a href="#Введение" data-toc-modified-id="Введение-1.1">Введение</a></span></li><li><span><a href="#Шаг-1.-Откройте-файл-с-данными-и-изучите-общую-информацию" data-toc-modified-id="Шаг-1.-Откройте-файл-с-данными-и-изучите-общую-информацию-1.2">Шаг 1. Откройте файл с данными и изучите общую информацию</a></span></li><li><span><a href="#Шаг-2.-Предобработка-данных" data-toc-modified-id="Шаг-2.-Предобработка-данных-1.3">Шаг 2. Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Переименование-столбцов" data-toc-modified-id="Переименование-столбцов-1.3.1">Переименование столбцов</a></span></li><li><span><a href="#Обработка-пропусков" data-toc-modified-id="Обработка-пропусков-1.3.2">Обработка пропусков</a></span><ul class="toc-item"><li><span><a href="#Столбец-days_employed" data-toc-modified-id="Столбец-days_employed-1.3.2.1">Столбец days_employed</a></span></li><li><span><a href="#Заполнение-пропусков" data-toc-modified-id="Заполнение-пропусков-1.3.2.2">Заполнение пропусков</a></span></li></ul></li><li><span><a href="#Замена-типа-данных" data-toc-modified-id="Замена-типа-данных-1.3.3">Замена типа данных</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-1.3.4">Обработка дубликатов</a></span><ul class="toc-item"><li><span><a href="#Столбец-children" data-toc-modified-id="Столбец-children-1.3.4.1">Столбец children</a></span></li><li><span><a href="#Столбец-days_employed" data-toc-modified-id="Столбец-days_employed-1.3.4.2">Столбец days_employed</a></span></li><li><span><a href="#Столбец-age" data-toc-modified-id="Столбец-age-1.3.4.3">Столбец age</a></span></li><li><span><a href="#Столбец-education" data-toc-modified-id="Столбец-education-1.3.4.4">Столбец education</a></span></li><li><span><a href="#Столбец-education_id" data-toc-modified-id="Столбец-education_id-1.3.4.5">Столбец education_id</a></span></li><li><span><a href="#Столбец--family_status" data-toc-modified-id="Столбец--family_status-1.3.4.6">Столбец  family_status</a></span></li><li><span><a href="#Столбец-family_status_id" data-toc-modified-id="Столбец-family_status_id-1.3.4.7">Столбец family_status_id</a></span></li><li><span><a href="#Столбец-gender" data-toc-modified-id="Столбец-gender-1.3.4.8">Столбец gender</a></span></li><li><span><a href="#Столбец-income_type" data-toc-modified-id="Столбец-income_type-1.3.4.9">Столбец income_type</a></span></li><li><span><a href="#Столбец-debt" data-toc-modified-id="Столбец-debt-1.3.4.10">Столбец debt</a></span></li><li><span><a href="#Столбец-total_income" data-toc-modified-id="Столбец-total_income-1.3.4.11">Столбец total_income</a></span></li><li><span><a href="#Столбец-purpose" data-toc-modified-id="Столбец-purpose-1.3.4.12">Столбец purpose</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-1.3.4.13">Обработка дубликатов</a></span></li></ul></li><li><span><a href="#Лемматизация" data-toc-modified-id="Лемматизация-1.3.5">Лемматизация</a></span></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-1.3.6">Категоризация данных</a></span><ul class="toc-item"><li><span><a href="#Категоризация-по-уровню-дохода" data-toc-modified-id="Категоризация-по-уровню-дохода-1.3.6.1">Категоризация по уровню дохода</a></span></li><li><span><a href="#Категоризация-по-целям-кредита" data-toc-modified-id="Категоризация-по-целям-кредита-1.3.6.2">Категоризация по целям кредита</a></span></li></ul></li></ul></li><li><span><a href="#Шаг-3.-Ответьте-на-вопросы" data-toc-modified-id="Шаг-3.-Ответьте-на-вопросы-1.4">Шаг 3. Ответьте на вопросы</a></span></li><li><span><a href="#Шаг-4.-Общий-вывод" data-toc-modified-id="Шаг-4.-Общий-вывод-1.5">Шаг 4. Общий вывод</a></span><ul class="toc-item"><li><span><a href="#1-этап.-Общее-изучение-входных-данных-от-банка." data-toc-modified-id="1-этап.-Общее-изучение-входных-данных-от-банка.-1.5.1">1 этап. Общее изучение входных данных от банка.</a></span></li><li><span><a href="#2-этап.-Подготовка-данных." data-toc-modified-id="2-этап.-Подготовка-данных.-1.5.2">2 этап. Подготовка данных.</a></span></li><li><span><a href="#3-этап.-Обработка-дубликатов." data-toc-modified-id="3-этап.-Обработка-дубликатов.-1.5.3">3 этап. Обработка дубликатов.</a></span></li><li><span><a href="#4-этап.-Лемматизация." data-toc-modified-id="4-этап.-Лемматизация.-1.5.4">4 этап. Лемматизация.</a></span></li><li><span><a href="#5-этап.-Категоризация-данных." data-toc-modified-id="5-этап.-Категоризация-данных.-1.5.5">5 этап. Категоризация данных.</a></span></li><li><span><a href="#6-этап.-Результаты-исследования." data-toc-modified-id="6-этап.-Результаты-исследования.-1.5.6">6 этап. Результаты исследования.</a></span></li><li><span><a href="#7-этап.-Общий-вывод." data-toc-modified-id="7-этап.-Общий-вывод.-1.5.7">7 этап. Общий вывод.</a></span>

## Введение  

 

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.

Результаты исследования будут учтены при построении модели **кредитного скоринга** — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

В ходе исследования надёжности заёмщиков перед нами стоят следующие **вопросы**:

- Наличие зависимости между количеством детей и возвратом кредита в срок.  
- Наличие зависимости между семейным положением и возвратом кредита в срок.  
- Наличие зависимости между уровнем дохода и возвратом кредита в срок.  
- Влияние целей кредита на соблюдение сроков его возврата.    

**Критерии** признания зависимости надежности заемщика от вышеуказанных признаков:

- % нарушений сроков возврата кредита существенным образом варьируется у заемщиков с разным количеством детей.  
- % нарушений сроков возврата кредита существенным образом варьируется у заемщиков с разным семейным статусом.
- % нарушений сроков возврата кредита существенным образом варьируется у заемщиков с разным доходом.
- % нарушений сроков возврата кредита существенным образом варьируется в зависимости от целей кредита.

*Уровень существенности* в рамках данного исследования принимается равным 5%.

Исследование надёжности заёмщиков включает в себя следующие **этапы**:
1. Общее изучение входных данных от банка.
2. Подготовка данных.
3. Обработка дубликатов.
4. Лемматизация.
5. Категоризация.
6. Результаты исследования (ответы на вопросы).
7. Общий вывод.

**Описание данных**:

*children* — количество детей в семье  
*days_employed* — общий трудовой стаж в днях  
*dob_years* — возраст клиента в годах  
*education* — уровень образования клиента  
*education_id* — идентификатор уровня образования  
*family_status* — семейное положение  
*family_status_id* — идентификатор семейного положения  
*gender* — пол клиента  
*income_type* — тип занятости  
*debt* — имел ли задолженность по возврату кредитов  
*total_income* — ежемесячный доход  
*purpose* — цель получения кредита  

## Шаг 1. Откройте файл с данными и изучите общую информацию

In [1]:
import pandas as pd
pd.set_option('display.float_format', '{:,.2f}'.format)

In [2]:
clients = pd.read_csv('/datasets/data.csv')

In [3]:
clients.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
children            21525 non-null int64
days_employed       19351 non-null float64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        19351 non-null float64
purpose             21525 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


In [4]:
clients.head(10)

children  days_employed  dob_years education  education_id  \
0         1      -8,437.67         42    высшее             0   
1         1      -4,024.80         36   среднее             1   
2         0      -5,623.42         33   Среднее             1   
3         3      -4,124.75         32   среднее             1   
4         0     340,266.07         53   среднее             1   
5         0        -926.19         27    высшее             0   
6         0      -2,879.20         43    высшее             0   
7         0        -152.78         50   СРЕДНЕЕ             1   
8         2      -6,929.87         35    ВЫСШЕЕ             0   
9         0      -2,188.76         41   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0    253,875.64   
1   женат / замужем                 0      F   сотрудник     0    112,080.01   
2   женат / замужем                 0      M   сотрудник     0    145,885.95   
3   женат / замужем                 0      M   сотрудник     0    267,628.55   
4  гражданский брак                 1      F   пенсионер     0    158,616.08   
5  гражданский брак                 1      M   компаньон     0    255,763.57   
6   женат / замужем                 0      F   компаньон     0    240,525.97   
7   женат / замужем                 0      M   сотрудник     0    135,823.93   
8  гражданский брак                 1      F   сотрудник     0     95,856.83   
9   женат / замужем                 0      M   сотрудник     0    144,425.94   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу  
5               покупка жилья  
6           операции с жильем  
7                 образование  
8       на проведение свадьбы  
9     покупка жилья для семьи

In [5]:
clients.describe()

children  days_employed  dob_years  education_id  family_status_id  \
count 21,525.00      19,351.00  21,525.00     21,525.00         21,525.00   
mean       0.54      63,046.50      43.29          0.82              0.97   
std        1.38     140,827.31      12.57          0.55              1.42   
min       -1.00     -18,388.95       0.00          0.00              0.00   
25%        0.00      -2,747.42      33.00          1.00              0.00   
50%        0.00      -1,203.37      42.00          1.00              0.00   
75%        1.00        -291.10      53.00          1.00              1.00   
max       20.00     401,755.40      75.00          4.00              4.00   

           debt  total_income  
count 21,525.00     19,351.00  
mean       0.08    167,422.30  
std        0.27    102,971.57  
min        0.00     20,667.26  
25%        0.00    103,053.15  
50%        0.00    145,017.94  
75%        0.00    203,435.07  
max        1.00  2,265,604.03

**Вывод**

- Входные данные представляют собой таблицу из 12 колонок и 21 525 строк.  
- Наименования столбцов преимущественно отражают суть содержащихся в них данных. Для улучшения восприятия столбец "dob_years" переименуем в "age".  
- Тип данных в столбцах  "days_employed" и "total_income" имеют тип float64. Для лучшей читаемости данных изменим тип на int64.  
- Исходные данные неидеальны:  
    - Столбцы "days_employed" и "total_income" содержат пропуски - 10% общего количества строк, что является существенным. Подход к заполнению пропусков - статистический, использование медианы.  
    - Столбец "children": есть отрицательные (-1) и нереальные (20) значения.
    - Столбец "days_employed": есть отрицательные значения и "ошибочные" значения (скорее всего являющиеся периодом работы в часах, а не днях).
    - Столбец "education": данные написаны в разных регистрах.

## Шаг 2. Предобработка данных

### Переименование столбцов

In [6]:
clients.rename(columns={"dob_years": "age"}, inplace=True)
clients.columns

Index(['children', 'days_employed', 'age', 'education', 'education_id',
       'family_status', 'family_status_id', 'gender', 'income_type', 'debt',
       'total_income', 'purpose'],
      dtype='object')

**Вывод**

Для улучшения восприятия заголовков таблицы переименовали столбец "dob_years" на "age".

### Обработка пропусков

На 1 этапе выявлено, что в столбцах "days_employed" и "total_income" есть пропуски.  Заполним пропуски медианными значениями соответствующих групп клиентов (по категориям столбца income_type).

#### Столбец days_employed

Столбец days_employed отражает трудовой стаж заемщика. Значения должны быть не отрицательными, целыми, выраженными в единых единицах измерения - количество дней.

In [7]:
# Диапазон данных в столбце days_employed
print('Минимальное значение трудового стажа:', '{:,.1f}'.format(clients['days_employed'].min()))
print('Максимальное значение трудового стажа:', '{:,.1f}'.format(clients['days_employed'].max()))


Минимальное значение трудового стажа: -18,388.9
Максимальное значение трудового стажа: 401,755.4


Размах значений существенный. Левая граница диапазона ниже 0, что противоречит сути данных. Правая граница диапазона имеет аномально высокие значения и скорее всего является показателем трудового стажа, выраженным в астрономических часах.

Для расчета и последующего заполнения пропусков медианными значениями произведем предобработку данных в столбце days_employed.

In [8]:
print('Количество отрицательных значений:', clients.loc[clients['days_employed'] < 0, 'days_employed'].count())
#days_minus = clients.loc[clients['days_employed'] < 0, 'days_employed'].count() / clients.shape[0]
print('Доля отрицательных значений трудового стажа: {:.1%}'.format(clients.loc[clients['days_employed'] < 0, 'days_employed'].count() / clients.shape[0]))

Количество отрицательных значений: 15906
Доля отрицательных значений трудового стажа: 73.9%


In [9]:
# Приведем значения в диапазон неотрицательных значений
clients['days_employed'] = clients['days_employed'].abs()
print('Доля отрицательных значений трудового стажа: {:.1%}'.format(clients.loc[clients['days_employed'] < 0, 'days_employed'].count() / clients.shape[0]))

Доля отрицательных значений трудового стажа: 0.0%


In [10]:
#Переведем значения стажа в часах в дни
print('Количество значений трудового стажа в часах:', clients.loc[clients['days_employed'] > 17166, 'days_employed'].count())
print('Доля значений трудового стажа в часах: {:.1%}'.format(clients.loc[clients['days_employed'] > 17166, 'days_employed'].count() / clients.shape[0]))

Количество значений трудового стажа в часах: 3447
Доля значений трудового стажа в часах: 16.0%


In [11]:
# Приведем значения трудового стажа к единицам измерения в днях
clients.loc[clients['days_employed'] > 17166, 'days_employed'] = clients.loc[clients['days_employed'] > 17166, 'days_employed'] / 24
print('Доля значений трудового стажа в часах: {:.1%}'.format(clients.loc[clients['days_employed'] > 17166, 'days_employed'].count() / clients.shape[0]))

Доля значений трудового стажа в часах: 0.0%


**Вывод**  

Столбец days_employed содержит отрицательные значения и значения трудового стажа, выраженные в часах.
Отрицательные значения устранены посредством приведения столбца к абсолютным значениям.
Для корректировки значений трудового стажа в часах применена следующая логика. Принимаем максимальный нормальный трудовой стаж размером 47 лет (для мужчин, работавших от 18 до 65 лет) или 17166 дней. Все значения, превышающие этот порог, переведены в дни путем деления на 24. 


#### Заполнение пропусков 

In [12]:
# Определим общее количество пропусков
clients.isna().sum()

children               0
days_employed       2174
age                    0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

In [13]:
clients.isna().mean()

children           0.00
days_employed      0.10
age                0.00
education          0.00
education_id       0.00
family_status      0.00
family_status_id   0.00
gender             0.00
income_type        0.00
debt               0.00
total_income       0.10
purpose            0.00
dtype: float64

In [14]:
# Группы клиентов по типу дохода
clients['income_type'].value_counts()

сотрудник          11119
компаньон           5085
пенсионер           3856
госслужащий         1459
предприниматель        2
безработный            2
в декрете              1
студент                1
Name: income_type, dtype: int64

In [15]:
# Заполним пропуски в столбце days_employed медианными значениями по группам income_type
for income in clients['income_type'].unique():
    median = clients.loc[clients['income_type'] == income, 'days_employed'].median()
    #print(income, median)
    clients.loc[(clients['days_employed'].isna()) & (clients['income_type'] == income), 'days_employed'] = median


In [16]:
# Заполним пропуски в столбце total_income медианными значениями по группам income_type
for income in clients['income_type'].unique():
    median = clients.loc[clients['income_type'] == income, 'total_income'].median()
    #print(income, median)
    clients.loc[(clients['total_income'].isna()) & (clients['income_type'] == income), 'total_income'] = median


In [17]:
clients.isna().sum()

children            0
days_employed       0
age                 0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

**Вывод**

Количество пропусков в столбцах "days_employed" и "total_income" существенно и составляет 10,1%. Удалять строки - некорректно. 

Пропуски в столбцах "days_employed" и "total_income" заполнены *медианными значениями*, определенными в группах клиентов по типу дохода (столбец "income_type"). 

### Замена типа данных

На 1 этапе выявлена необходимость замены типа данных в столбцах "days_employed" и "total_income" с float64 на int64.

In [18]:
#Изменение типа данных в столбце days_employed
clients['days_employed'] = clients['days_employed'].astype('int')
clients['days_employed']

0         8437
1         4024
2         5623
3         4124
4        14177
         ...  
21520     4529
21521    14330
21522     2113
21523     3112
21524     1984
Name: days_employed, Length: 21525, dtype: int64

In [19]:
#Изменение типа данных в столбце total_income
clients['total_income'] = clients['total_income'].astype('int')
clients['total_income']

0        253875
1        112080
2        145885
3        267628
4        158616
          ...  
21520    224791
21521    155999
21522     89672
21523    244093
21524     82047
Name: total_income, Length: 21525, dtype: int64

**Вывод**

Произведена замена типа данных в столбцах "days_employed" и "total_income" с float64 на int64, что улучшает восприятие в процессе анализа данных.

### Обработка дубликатов

Проанализируем данные в каждом столбце исходной таблицы на предмет наличия дубликатов, аномальных значений. При необходимости произведем преобразования.

In [20]:
clients.columns

Index(['children', 'days_employed', 'age', 'education', 'education_id',
       'family_status', 'family_status_id', 'gender', 'income_type', 'debt',
       'total_income', 'purpose'],
      dtype='object')

#### Столбец children

Столбец children содержит информацию о количестве детей заемщиков. Значения должны быть не отрицательными, целыми, находиться в диапазоне [1:10].

In [21]:
# Значения в столбце children
clients['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

In [22]:
# Количество и доля аномальных значений в столбце children
child_minus = clients.loc[clients['children'] == -1,'children'].count()
d_child_minus = child_minus / len(clients['children'])
print('Количество значений "-1": ', child_minus)
print('Доля значений "-1": ''{:.1%}'.format(d_child_minus))

Количество значений "-1":  47
Доля значений "-1": 0.2%


In [23]:
# Количество и доля аномальных значений в столбце children
child_twenty = clients.loc[clients['children'] == 20,'children'].count()
d_child_twenty = child_twenty / len(clients['children'])
print('Количество значений "20": ', child_minus)
print('Доля значений "20": ''{:.1%}'.format(d_child_twenty))

Количество значений "20":  47
Доля значений "20": 0.4%


In [24]:
# Замена значений "-1" и "20" на медианные значения в столбце children
median_children = clients['children'].median()
#median_children
clients.loc[clients['children'] == -1, 'children'] = median_children
clients.loc[clients['children'] == 20, 'children'] = median_children

In [25]:
clients['children'].value_counts()

0.00    14272
1.00     4818
2.00     2055
3.00      330
4.00       41
5.00        9
Name: children, dtype: int64

**Вывод**

Столбец children содержит некорректные значения "-1" и "20". Количество и удельный вес аномальных значений в общем объеме данных невелики - 47 (0,2%) и 76 (0,4%) соответственно. Это позволяет без нарушения качества данных произвести замену на медианные значения группы.

#### Столбец days_employed

Столбец days_employed предобработан ранее на этапе обработки пропусков.

#### Столбец age


Столбец age отражает возраст заемщика и соответственной должен содержать положительные значения, больше 18.

In [26]:
clients['age'].unique()

array([42, 36, 33, 32, 53, 27, 43, 50, 35, 41, 40, 65, 54, 56, 26, 48, 24,
       21, 57, 67, 28, 63, 62, 47, 34, 68, 25, 31, 30, 20, 49, 37, 45, 61,
       64, 44, 52, 46, 23, 38, 39, 51,  0, 59, 29, 60, 55, 58, 71, 22, 73,
       66, 69, 19, 72, 70, 74, 75])

In [27]:

print('Количество значений возраста "0":', clients.loc[clients['age'] == 0, 'age'].count())
d_age_0 = clients.loc[clients['age'] == 0, 'age'].count() / len(clients['age'])
print('Доля значений возраста "0": {:.1%}'.format(d_age_0))

Количество значений возраста "0": 101
Доля значений возраста "0": 0.5%


In [28]:
median_age = clients['age'].median()
median_age

42.0

In [29]:
age_income_type = clients.loc[clients['age'] == 0, 'income_type']
age_income_type.head(10)

99        пенсионер
149       сотрудник
270       сотрудник
578       пенсионер
1040      компаньон
1149      компаньон
1175      пенсионер
1386    госслужащий
1890      сотрудник
1898      пенсионер
Name: income_type, dtype: object

Количество и доля некорректных значений возраста заемщика несущественны и составляют 101 и 0,5% соответственно. Медианное значение возраста составляет 42 года. Замена нулевых значений на медианные по группе в целом несправедлива, так как заведомо разновозрастные группы студент и пенсионер получат одинаковое значение.  С учетом этого заменим нулевые позиции на медианные значения возраста по группам, исходя из типа занятости.

In [30]:
for income in clients['income_type'].unique():
    median_age = clients.loc[clients['income_type'] == income, 'age'].median()
    #print(income, median_age)
    clients.loc[(clients['age'] == 0) & (clients['income_type'] == income), 'age'] = median_age

In [31]:
print('Доля значений возраста "0": {:.1%}'.format(clients.loc[clients['age'] == 0, 'age'].count() / len(clients['age'])))

Доля значений возраста "0": 0.0%


In [32]:
print('Количество значений возраста "меньше 18":', clients.loc[clients['age'] < 18, 'age'].count())

Количество значений возраста "меньше 18": 0


**Вывод**  
Столбец age, отражающий возраст заемщика, содержит некорректные значения, равные 0. Доля таких значений в выборке несущественна (0,5%), подобные значения заменены на медианные значения возраста по группам, исходя из типа занятости.


#### Столбец education

Столбец education отражает образование заемщика. Данные должны быть написаны без орфографических ошибок, в одном регистре.

In [33]:
clients['education'].unique()

array(['высшее', 'среднее', 'Среднее', 'СРЕДНЕЕ', 'ВЫСШЕЕ',
       'неоконченное высшее', 'начальное', 'Высшее',
       'НЕОКОНЧЕННОЕ ВЫСШЕЕ', 'Неоконченное высшее', 'НАЧАЛЬНОЕ',
       'Начальное', 'Ученая степень', 'УЧЕНАЯ СТЕПЕНЬ', 'ученая степень'],
      dtype=object)

In [34]:
# Преобразование столбца к нижнему регистру
clients['education'] = clients['education'].str.lower()

In [35]:
print(clients['education'].unique())

['высшее' 'среднее' 'неоконченное высшее' 'начальное' 'ученая степень']


**Вывод**  
Столбец education предобработан - данные записаны в одном регистре, без орфографических ошибок.

#### Столбец education_id

In [36]:
print(clients['education_id'].unique())

[0 1 2 3 4]


**Вывод**  
Столбец education_id предобработки не требует

#### Столбец  family_status

Столбец family_status отражает семейный статус заемщика. Данные должны быть написаны без орфографических ошибок, в одном регистре.

In [37]:
print(clients['family_status'].unique())

['женат / замужем' 'гражданский брак' 'вдовец / вдова' 'в разводе'
 'Не женат / не замужем']


In [38]:
clients['family_status'] = clients['family_status'].str.lower()

In [39]:
print(clients['family_status'].unique())

['женат / замужем' 'гражданский брак' 'вдовец / вдова' 'в разводе'
 'не женат / не замужем']


**Вывод**  
Столбец family_status предобработан - данные записаны в одном регистре, без орфографических ошибок.

#### Столбец family_status_id

In [40]:
print(clients['family_status_id'].unique())

[0 1 2 3 4]


**Вывод**  
Столбец family_status_id предобработки не требует

#### Столбец gender

Столбец gender отражает пол заемщика и для целей данного исследования принимает два значения - F (женщина) и M (мужчина).

In [41]:
print(clients['gender'].value_counts())

F      14236
M       7288
XNA        1
Name: gender, dtype: int64


In [42]:
clients.loc[clients['gender'] == 'XNA']

children  days_employed   age            education  education_id  \
10701      0.00           2358 24.00  неоконченное высшее             2   

          family_status  family_status_id gender income_type  debt  \
10701  гражданский брак                 1    XNA   компаньон     0   

       total_income               purpose  
10701        203905  покупка недвижимости

В гендерной структуре заемщиков преобладают женщины (66%). Высока вероятность того, что заемщик с типом гендера "XNA" - это женщина (F). Произведем замену.

In [43]:
clients.loc[clients['gender'] == 'XNA', 'gender'] = 'F'

In [44]:
print(clients['gender'].value_counts())

F    14237
M     7288
Name: gender, dtype: int64


**Вывод**  
Столбец gender содержал неопределенный тип гендера "XNA". С учетом гендерной структуры заемщиков данное значение заменено на F (женщина).

#### Столбец income_type

Столбец income_type отражает тип дохода заемщика. Значения должны быть записаны в одном регистре, без орфографических ошибок.

In [45]:
clients['income_type'].unique()

array(['сотрудник', 'пенсионер', 'компаньон', 'госслужащий',
       'безработный', 'предприниматель', 'студент', 'в декрете'],
      dtype=object)

**Вывод**  
Столбец income_type предобработки не требует.

#### Столбец debt

Столбец debt отражает, имел ли заемщик задолженность по возврату кредитов. Значения 0 и 1. 0 - задолженности не было, 1 - задолженность была.

In [46]:
print(clients['debt'].unique())

[0 1]


**Вывод**  
Столбец debt предобработки не требует.

#### Столбец total_income

Столбец total_income отражает ежемесячный доход заемщика.

In [47]:
print('Минимальный ежемесячный доход:', clients['total_income'].min())
print('Максимальный ежемесячный доход:', clients['total_income'].max())

Минимальный ежемесячный доход: 20667
Максимальный ежемесячный доход: 2265604


**Вывод**  
Столбец total_income предобработки не требует.

#### Столбец purpose

Столбец purpose отражает цель получения кредита. Данные должны быть написаны в одном регистре.

In [48]:
clients['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Вывод**  
Столбец purpose заполнен в едином регистре. Скорее всего цель формулировали сами заемщики, поэтому много семантически близких конструкций, но нет единообразия. Требуется дополнительный анализ и обработка.

#### Обработка дубликатов

In [49]:
print(clients.duplicated().sum()) 

71


In [50]:
# Удаление дубликатов
clients = clients.drop_duplicates().reset_index(drop=True)
print(clients.duplicated().sum()) 

0


**Вывод**  
На текущем этапе удалены дубликаты в исходных данных. Аномальные количественные и качественные значения обработаны в соответствии с критериями данных для каждого столбца.

### Лемматизация

Одним из вопросов, на который мы пытаемся ответить в рамках данного исследования, *влияют ли разные цели кредита на соблюдение сроков его возврата*. На предыдущем этапе в процессе предобработки столбца *purpose* было выявлено разнообразие формулировок целей. С помощью *лемматизации* попробуем выявить основные цели кредитов и сформировать по этому признаку несколько групп.


In [51]:
from pymystem3 import Mystem
m = Mystem() 

In [52]:
# Преобразование столбца purpose в строку
str_purpose = ' '.join(clients['purpose'])
#str_purpose

In [53]:
# Лемматизация purpose
lemmas = m.lemmatize(str_purpose)
#lemmas

In [54]:
# Подсчет лемм
from collections import Counter
count_lemmas = Counter(lemmas)
print(count_lemmas)

Counter({' ': 55023, 'недвижимость': 6351, 'покупка': 5897, 'жилье': 4460, 'автомобиль': 4306, 'образование': 4013, 'с': 2918, 'операция': 2604, 'свадьба': 2324, 'свой': 2230, 'на': 2222, 'строительство': 1878, 'высокий': 1374, 'получение': 1314, 'коммерческий': 1311, 'для': 1289, 'жилой': 1230, 'сделка': 941, 'дополнительный': 906, 'заниматься': 904, 'подержать': 853, 'проведение': 768, 'сыграть': 765, 'сдача': 651, 'семья': 638, 'собственный': 635, 'со': 627, 'ремонт': 607, 'приобретение': 461, 'профильный': 436, 'подержанный': 111, '\n': 1})


In [55]:
# Преобразование результата работы Counter в датафрейм
count_lemmas_df = pd.DataFrame.from_dict(count_lemmas, orient ='index').reset_index()
#count_lemmas_df

In [56]:
# Переименование столбцов датафрейма count_lemmas_df
count_lemmas_df = count_lemmas_df.rename(columns={'index':'word', 0:'repeat'})
# Сортировка по частоте лемм
count_lemmas_df = count_lemmas_df.sort_values(by = 'repeat', ascending = False)
# Удалим значения 'пробел'
index = count_lemmas_df[count_lemmas_df['word'] == ' '].index
count_lemmas_df.drop(index, inplace = True)
# Расчет доли лемм в общем количестве
count_lemmas_df['d_repeat'] = count_lemmas_df['repeat'] / count_lemmas_df['repeat'].sum()*100
print(count_lemmas_df)

              word  repeat  d_repeat
15    недвижимость    6351     11.54
0          покупка    5897     10.72
2            жилье    4460      8.11
4       автомобиль    4306      7.83
6      образование    4013      7.29
10               с    2918      5.30
9         операция    2604      4.73
8          свадьба    2324      4.22
21            свой    2230      4.05
11              на    2222      4.04
18   строительство    1878      3.41
26         высокий    1374      2.50
25       получение    1314      2.39
16    коммерческий    1311      2.38
13             для    1289      2.34
17           жилой    1230      2.24
24          сделка     941      1.71
5   дополнительный     906      1.65
23      заниматься     904      1.64
20       подержать     853      1.55
12      проведение     768      1.40
7          сыграть     765      1.39
29           сдача     651      1.18
14           семья     638      1.16
19     собственный     635      1.15
22              со     627      1.14
3

**Вывод**  
Произведена лемматизация столбца *purpose*. Леммы, которые станут основой для дальнейшей категоризации:
- недвижимость
- покупка
- жилье
- автомобиль
- образование
- свадьба

### Категоризация данных

Для дальнейшего анализа данных и ответа на вопросы исследования необходимо произвести категоризацию данных по следующим признакам:
- уровень дохода;
- цель получения кредита.

#### Категоризация по уровню дохода

Для категоризации по уровню дохода (столбец total_income) распределим заемщиков по следующим группам:
- низкий доход (low_income);
- доход ниже среднего (below_average_income);
- средний доход (average_income);
- доход выше среднего (above_average_income);
- высокий доход (high_income).  

Используем распределение по квинтилям - 20%-ные группы заемщиков, ранжированных по мере возрастания доходов. Данный подход является достаточно распространенным в социально-экономической статистике (используется Росстатом для анализа денежных доходов населения (https://rosstat.gov.ru/folder/13723?print=1).

In [57]:
bin_labels_5 = ['низкий доход', 'доход ниже среднего', 'средний доход', 'доход выше среднего', 'высокий доход']
#bin_labels_5 = ['low_income', 'below_average_income', 'average_income', 'above_average_income', 'high_income']
clients['income_group'] = pd.qcut(clients['total_income'], q=[0, .2, .4, .6, .8, 1], labels=bin_labels_5)

In [58]:
clients.head(10)

children  days_employed   age education  education_id     family_status  \
0      1.00           8437 42.00    высшее             0   женат / замужем   
1      1.00           4024 36.00   среднее             1   женат / замужем   
2      0.00           5623 33.00   среднее             1   женат / замужем   
3      3.00           4124 32.00   среднее             1   женат / замужем   
4      0.00          14177 53.00   среднее             1  гражданский брак   
5      0.00            926 27.00    высшее             0  гражданский брак   
6      0.00           2879 43.00    высшее             0   женат / замужем   
7      0.00            152 50.00   среднее             1   женат / замужем   
8      2.00           6929 35.00    высшее             0  гражданский брак   
9      0.00           2188 41.00   среднее             1   женат / замужем   

   family_status_id gender income_type  debt  total_income  \
0                 0      F   сотрудник     0        253875   
1                 0      F   сотрудник     0        112080   
2                 0      M   сотрудник     0        145885   
3                 0      M   сотрудник     0        267628   
4                 1      F   пенсионер     0        158616   
5                 1      M   компаньон     0        255763   
6                 0      F   компаньон     0        240525   
7                 0      M   сотрудник     0        135823   
8                 1      F   сотрудник     0         95856   
9                 0      M   сотрудник     0        144425   

                      purpose         income_group  
0               покупка жилья        высокий доход  
1     приобретение автомобиля  доход ниже среднего  
2               покупка жилья        средний доход  
3  дополнительное образование        высокий доход  
4             сыграть свадьбу        средний доход  
5               покупка жилья        высокий доход  
6           операции с жильем        высокий доход  
7                 образование        средний доход  
8       на проведение свадьбы         низкий доход  
9     покупка жилья для семьи        средний доход

In [59]:
#Проверка распределения
clients['income_group'].value_counts()

высокий доход          4291
доход выше среднего    4291
доход ниже среднего    4291
низкий доход           4291
средний доход          4290
Name: income_group, dtype: int64

In [60]:
#results, bin_edges = pd.qcut(clients['income_group'], q=[0, .2, .4, .6, .8, 1], labels=bin_labels_5, retbins=True)
#results_table = pd.DataFrame(zip(bin_edges, bin_labels_5), columns=['Threshold', 'Tier'])

**Вывод**  
Заемщики разделены на 5 групп в соответствии с уровнем дохода:
- низкий доход (low_income);
- доход ниже среднего (below_average_income);
- средний доход (average_income);
- доход выше среднего (above_average_income);
- высокий доход (high_income).  
 
 Распределение произведено по квинтилям - 20%-ные группы заемщиков, ранжированных по мере возрастания доходов.

#### Категоризация по целям кредита

Для категоризации целей распределим заемщиков по следующим группам:
- жилая недвижимость;
- коммерческая недвижимость;
- автомобиль;
- образование;
- свадьба;
- прочие.  

Распределение опирается на результаты *лемматизации столбца purpose* и учитывает следующие допущения:
- если в целях кредита не обозначен тип недвижимости, то относим такую цель к категории "жилая недвижимость";
- категория "жилая недвижимость" содержит все заявленные типы операций с ней (строительство, приобретение, ремонт).

In [61]:
def purpose_cat(row):
    '''
    Функция распределения заемщиков по целям
    '''
    purpose = row['purpose']
    lemmas = m.lemmatize(purpose)
    
    if 'автомобиль' in lemmas:
        return 'автомобиль'
    if 'свадьба' in lemmas:
        return 'свадьба'
    if 'образование' in lemmas:
        return 'образование'
    if 'недвижимость' in lemmas and 'коммерческий' in lemmas:
        return 'коммерческая недвижимость'
    if 'жилье' in lemmas or 'жилой' in lemmas:
        return 'жилая недвижимость'
    if 'недвижимость' in lemmas and not 'коммерческий' in lemmas:
        return 'жилая недвижимость'
    return 'прочие'

In [62]:
# Создание столбца purpose_group c категориями доходов
clients['purpose_group'] = clients.apply(purpose_cat, axis = 1)
clients. head(10)

children  days_employed   age education  education_id     family_status  \
0      1.00           8437 42.00    высшее             0   женат / замужем   
1      1.00           4024 36.00   среднее             1   женат / замужем   
2      0.00           5623 33.00   среднее             1   женат / замужем   
3      3.00           4124 32.00   среднее             1   женат / замужем   
4      0.00          14177 53.00   среднее             1  гражданский брак   
5      0.00            926 27.00    высшее             0  гражданский брак   
6      0.00           2879 43.00    высшее             0   женат / замужем   
7      0.00            152 50.00   среднее             1   женат / замужем   
8      2.00           6929 35.00    высшее             0  гражданский брак   
9      0.00           2188 41.00   среднее             1   женат / замужем   

   family_status_id gender income_type  debt  total_income  \
0                 0      F   сотрудник     0        253875   
1                 0      F   сотрудник     0        112080   
2                 0      M   сотрудник     0        145885   
3                 0      M   сотрудник     0        267628   
4                 1      F   пенсионер     0        158616   
5                 1      M   компаньон     0        255763   
6                 0      F   компаньон     0        240525   
7                 0      M   сотрудник     0        135823   
8                 1      F   сотрудник     0         95856   
9                 0      M   сотрудник     0        144425   

                      purpose         income_group       purpose_group  
0               покупка жилья        высокий доход  жилая недвижимость  
1     приобретение автомобиля  доход ниже среднего          автомобиль  
2               покупка жилья        средний доход  жилая недвижимость  
3  дополнительное образование        высокий доход         образование  
4             сыграть свадьбу        средний доход             свадьба  
5               покупка жилья        высокий доход  жилая недвижимость  
6           операции с жильем        высокий доход  жилая недвижимость  
7                 образование        средний доход         образование  
8       на проведение свадьбы         низкий доход             свадьба  
9     покупка жилья для семьи        средний доход  жилая недвижимость

**Вывод**  
Произведена категоризация целей кредита (столбец *purpose*), опираясь на результаты лемматизации. Выделено 5 основных категорий (жилая недвижимость; коммерческая недвижимость; автомобиль; образование; свадьба), оставшиеся цели объединены в категорию "прочие". 

## Шаг 3. Ответьте на вопросы

- Есть ли зависимость между наличием детей и возвратом кредита в срок?

In [63]:
children_pivot = clients.pivot_table(index = 'children', values = 'debt',  aggfunc = ['sum','count'])
children_pivot.columns = ['Количество невозвратов' , 'Количество заемщиков']


In [64]:
children_pivot['% невозврата'] = children_pivot['Количество невозвратов'] /children_pivot['Количество заемщиков']
children_pivot = children_pivot.sort_values(by = '% невозврата', ascending = False)
children_pivot.style.format({'% невозврата':'{:.2%}'})

**Вывод**  
Рассчитана доля нарушений срока возврата кредитов (столбец "% невозврата") для заемщиков, сгруппированных по количеству детей.
Данный показатель варьируется по группам незначительно от 7,5% до 9,8% (на 2,3%%). Самый низкий процент невозвратов в группе заемщиков без детей - 7,5%.
Заемщики с 5 детьми в предоставленных данных всегда возвращали кредит в срок. Однако ввиду незначительного количества данного типа заемщиков в общей структуре - 0,04% - эта ситуация не показательна.  
Вышесказанное дает нам основания считать, что **факт наличия детей у заемщика, а также их количество практически не влияют на платежную дисциплину**.  


- Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [65]:
family_status_pivot = clients.pivot_table(index = 'family_status', values = 'debt',  aggfunc = ['sum','count'])
family_status_pivot.columns = ['Количество невозвратов' , 'Количество заемщиков']


In [66]:
family_status_pivot['% невозврата'] = family_status_pivot['Количество невозвратов'] /family_status_pivot['Количество заемщиков']
family_status_pivot = family_status_pivot.sort_values(by = '% невозврата', ascending = False)
family_status_pivot.style.format({'% невозврата':'{:.2%}'})

**Вывод**

Рассчитана доля нарушений срока возврата кредита (столбец "% невозврата") для заемщиков с разным семейным статусом.
Данный показатель варьируется по группам незначительно от 6,6% до 9,8% (на 3,2%%). Самый низкий процент невозврата в группе "вдовец/вдова" - 6,6%. Самый высокий показатель по нарушениям в группе "не женат / не замужем".  
Вышесказанное дает нам основания считать, что **семейный статус оказывает слабое влияние на платежную дисциплину**.

- Есть ли зависимость между уровнем дохода и возвратом кредита в срок

In [67]:
income_group_pivot = clients.pivot_table(index = 'income_group', values = 'debt',  aggfunc = ['sum','count'])
income_group_pivot.columns = ['Количество невозвратов' , 'Количество заемщиков']


In [68]:
income_group_pivot['% невозврата'] = income_group_pivot['Количество невозвратов'] /income_group_pivot['Количество заемщиков']
income_group_pivot = income_group_pivot.sort_values(by = '% невозврата', ascending = False)
income_group_pivot.style.format({'% невозврата':'{:.2%}'})


**Вывод**

Рассчитана доля нарушений срока возврата кредитов для каждой из групп заемщиков по уровню дохода.  
Данный показатель варьируется по группам незначительно от 7,0% до 8,7% (на 0,9%%). Самый низкий процент неплатежей в группе заемщиков с высоким доходом - 7,0 %.   
Вышесказанное дает нам основания считать, что **уровень дохода практически не оказывает влияния на платежную дисциплину**.

- Как разные цели кредита влияют на его возврат в срок?

In [69]:
purpose_group_pivot = clients.pivot_table(index = 'purpose_group', values = 'debt',  aggfunc = ['sum','count'])
purpose_group_pivot.columns = ['Количество невозвратов' , 'Количество заемщиков']

In [70]:
purpose_group_pivot['% невозврата'] = purpose_group_pivot['Количество невозвратов'] /purpose_group_pivot['Количество заемщиков']
purpose_group_pivot = purpose_group_pivot.sort_values(by = '% невозврата', ascending = False)
purpose_group_pivot.style.format({'% невозврата':'{:.2%}'})

**Вывод**

Рассчитана доля нарушений срока возврата кредитов по группам заемщиков в зависимости от цели кредита.
Данный показатель варьируется по группам незначительно от 7,2% до 9,4% (на 2,2%%).  
Целевые кредиты (автокредит и образовательный) возвращают чуть хуже остальных (9,4% т 9,2% соответственно).
Вышесказанное дает нам основания считать, что **цель кредита оказывает слабое влияние на платежную дисциплину**.

## Шаг 4. Общий вывод

Проведено исследование надежности заемщиков. Проанализировано влияние различных факторов на соблюдение условий кредитного договора в части возврата сумм кредита:
- количество детей;
- семейный статус;
- уровень дохода;
- цель кредита.  

Исследование включает в себя ряд этапов:
1. Общее изучение входных данных от банка.
2. Подготовка данных.
3. Обработка дубликатов.
4. Лемматизация.
5. Категоризация.
6. Результаты исследования (ответы на вопросы).
7. Общий вывод.  

### 1 этап. Общее изучение входных данных от банка. 

На первом этапе выявлены общие характеристики данных, предоставленных банком:
- большая таблица (12 колонок, 21 525 строк);
- наименования столбцов отражают, что есть все параметры, необходимые для дальнейших расчетов;
- исходные данные требуют предобработки (работы с пропусками, замена типа данных, исправления аномальных значений).

### 2 этап. Подготовка данных.

*2.1. Переименование столбцов.*  
Для улучшения восприятия заголовков таблицы переименован столбец "dob_years" на "age".

*2.2. Обработка пропусков.*  
В столбцах "days_employed" и "total_income" обнаружены пропуски. Количество пропусков существенно - 10,1%. Исключение пропусков может привести к искажению информации и последующих выводов.
Пропуски в столбцах "days_employed" и "total_income" заполнены *медианными значениями,* определенными в группах клиентов по типу дохода (столбец "income_type").

Для столбца *days_employed*, содержащего информацию о трудовом стаже заемщика, до заполнения пропусков проведено преобразование отрицательных значений и аномально высоких показателей.
*Отрицательные значения* устранены посредством приведения столбца к абсолютным значениям.  
*Аномально высокие значения* - скорее всего показатели трудового стажа в часах. Наличие в столбце days_employed показателей в разных единицах измерения - дни и часы - вероятнее всего связано с использованием выгрузок из разных учетных систем.
Для корректировки значений трудового стажа в часах применена следующая логика. Принимаем максимальный нормальный трудовой стаж размером 47 лет (для мужчин, работавших от 18 до 65 лет) или 17166 дней. Все значения, превышающие этот порог, переведены в дни путем деления на 24. 




*2.3. Замена типа данных.*  
Произведена замена типа данных в столбцах "days_employed" и "total_income" с float64 на int64, что улучшает восприятие в процессе анализа данных.


### 3 этап. Обработка дубликатов.

На 3-м этапе проанализированы данные в каждом столбце исходной таблицы на предмет наличия дубликатов, аномальных значений. Произведены следующие преобразования.  

*Столбец children.*  
Столбец children содержит информацию о количестве детей заемщика. Значения должны быть не отрицательными, целыми, находиться в диапазоне [1:10].  
Произведена замена некорректных значений "-1" и "20" на медианные значения группы.  

*Столбец days_employed.*  
Предобработка проведена на этапе заполнения пропусков.

*Столбец age.*  
Столбец age, отражающий возраст заемщика, содержит некорректные значения, равные 0. Доля таких значений в выборке несущественна (0,5%), подобные значения заменены на медианные значения данной группы.  

*Столбец education.*  
Столбец education, отражающий образование заемщика, предобработан - данные записаны в одном регистре, без орфографических ошибок.  

*Столбец education_id.*  
Столбец education_id содержит идентификатор уровня образования, предобработки не требует.

*Столбец family_status.*  
Столбец family_status отражает семейный статус заемщика, предобработан - данные записаны в одном регистре, без орфографических ошибок.

*Столбец family_status_id.*  
Столбец family_status_id содержит идентификатор семейного положения и предобработки не требует

*Столбец income_type.*  
Столбец income_type отражает тип дохода заемщика,предобработки не требует.

*Столбец debt.*  
Столбец debt отражает, имел ли заемщик задолженность по возврату кредитов, предобработки не требует.

*Столбец total_income.*  
Столбец total_income отражает ежемесячный доход заемщика, предобработки не требует.

*Столбец purpose.*  
Столбец purpose отражает цель получения кредита, предобработки не требует.

### 4 этап. Лемматизация.

На данном этапе проведена лемматизация *столбца purpose*. Выявлены леммы, которые лягут в основу категоризации целей кредита.
- недвижимость,
- покупка,
- жилье,
- автомобиль,
- образование,
- свадьба.

###  5 этап. Категоризация данных.

На данном этапе произведена категоризация заемщиков по следующим признакам:  
- уровень дохода (столбец total_income);
- цель получения кредита (столбец purpose).

При категоризации заемщиков по *уровню дохода (столбец total_income)* используется распределение по квинтилям - 20%-ные группы заемщиков, ранжированных по мере возрастания доходов. Данный подход  взят как наиболее универсальный в социально-экономической статистике.  
В результате заемщики разделены на *5 групп* в соответствии с уровнем дохода:
- низкий доход (low_income);
- доход ниже среднего (below_average_income);
- средний доход (average_income);
- доход выше среднего (above_average_income);
- высокий доход (high_income).  
 
 
При категоризации заемщиков *по целям кредита* используются результаты лемматизации столбца purpose. В результате заемщики распределены по следующим целевым группам:  
- жилая недвижимость;
- коммерческая недвижимость;
- автомобиль;
- образование;
- свадьба;
- прочие.


### 6 этап. Результаты исследования.

Для исследования влияния (зависимости) разных характеристик заемщика (наличие детей, семейное положение , уровень дохода, цель кредита) на факт погашения кредита в срок применяется следующий подход:  
- заемщики распределены по группам в рамках каждого исследуемого признака (наличие детей, семейное положение , уровень дохода, цель кредита);
- для каждой группы произведен  расчет % невозврата кредита;
- произведено межгрупоовое сравнение показателя;
- при наличиии существенной вариативности показателя признается влияние признака на факт погашения кредита в срок.

Уровень существенности в рамках данного исследования принимается равным 5%.


- Есть ли зависимость между наличием детей и возвратом кредита в срок?  

Вариативность % невозврата между группами заемщиков с разным количеством детей - 2,3%%. 
Факт наличия детей у заемщика, а также их количество практически не влияют на платежную дисциплину.

- Есть ли зависимость между семейным положением и возвратом кредита в срок?  

Вариативность % невозврата между группами заемщиков с разным семейным статусом - 3,2%%. 
Семейный статус оказывает слабое влияние на платежную дисциплину.

- Есть ли зависимость между уровнем дохода и возвратом кредита в срок?  

Вариативность % невозврата между группами заемщиков с разным уровнем дохода - 0,9%%. 
Уровень дохода практически не оказывает влияния на платежную дисциплину.

- Как разные цели кредита влияют на его возврат в срок?  

Вариативность % невозврата между группами заемщиков с разными целями кредита - 2,2%%. 
Цель кредита оказывает слабое влияние на платежную дисциплину.

### 7 этап. Общий вывод.

Проведенный анализ наглядно показывает отсутствие существенного влияния количеcтва детей, семейного положения, уровня дохода и целей кредита на факт погашения кредита в срок.